commit b674cc5ba47970b3ef0f81ead4da508cbad285fb
Author: Christian Glindkamp <christian.glindkamp@taskit.de>
Date:   Tue Jul 26 16:29:25 2011 +0200

    mmc: atmel-mci: use flush_kernel_dcache_page()
    
    flush_dcache_page() results in IRQs being enabled due to the
    flush_dcache_mmap_lock() (as pointed out Russel King) thereby causing:
     WARNING: at kernel/irq/handle.c:130 handle_irq_event_percpu+0x70/0x194()
     irq 29 handler atmci_interrupt+0x0/0x64c enabled interrupts
    
    Signed-off-by: Christian Glindkamp <christian.glindkamp@taskit.de>

diff --git a/drivers/mmc/host/atmel-mci.c b/drivers/mmc/host/atmel-mci.c
index aa8039f..6afa9ba 100644
--- a/drivers/mmc/host/atmel-mci.c
+++ b/drivers/mmc/host/atmel-mci.c
@@ -1387,7 +1387,7 @@ static void atmci_read_data_pio(struct atmel_mci *host)
 			nbytes += 4;
 
 			if (offset == sg->length) {
-				flush_dcache_page(sg_page(sg));
+				flush_kernel_dcache_page(sg_page(sg));
 				host->sg = sg = sg_next(sg);
 				if (!sg)
 					goto done;
@@ -1400,7 +1400,7 @@ static void atmci_read_data_pio(struct atmel_mci *host)
 			memcpy(buf + offset, &value, remaining);
 			nbytes += remaining;
 
-			flush_dcache_page(sg_page(sg));
+			flush_kernel_dcache_page(sg_page(sg));
 			host->sg = sg = sg_next(sg);
 			if (!sg)
 				goto done;
